packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -586,17 +612,32 @@ describe('KeyringController', () => {
               },
             );
           });
-        });
 
-        cacheEncryptionKey &&
-          it('should set encryptionKey and encryptionSalt in state', async () => {
-            // TODO: Either fix this lint violation or explain why it's necessary to ignore.
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            withController({ cacheEncryptionKey }, async ({ initialState }) => {
-              expect(initialState.encryptionKey).toBeDefined();
-              expect(initialState.encryptionSalt).toBeDefined();
-            });
+          it('should remove encryptionKey and encryptionSalt from state when cacheEncryptionKey is false', async () => {
+            await withController(
+              { skipVaultCreation: true },
```

Comment:
This test is under the `when there is an existing vault` describe block, so we would not want to skip initial vault creation here

Reply:
Oh and we'll want to preserve the `cacheEncryptionKey` option as well.

And as another nit, we can run this test conditionally (only when `cacheEncryptionKey` is false) for the same reasons as above